src/membership/members.rs

```diff
@@ -287,6 +308,38 @@ decl_module! {
             }
         }
 
+        pub fn set_controller_key(origin, controller: T::AccountId) {
+            let master = ensure_signed(origin)?;
+            let member_id = Self::ensure_is_member_primary_account(&master)?;
+
+            let mut profile = Self::ensure_profile(member_id)?;
+
+            // ensure new key is not used by someone else (master or controller)
+            ensure!(!<MemberIdByControllerAccountId<T>>::exists(&controller), "account already paired with member");
+            ensure!(!<MemberIdByAccountId<T>>::exists(&controller), "account already paired with member");
+
+            <MemberIdByControllerAccountId<T>>::remove(&profile.controller_account);
+            <MemberIdByControllerAccountId<T>>::insert(&controller, member_id);
+            profile.controller_account = controller.clone();
+            <MemberProfile<T>>::insert(member_id, profile);
```

Comment:
Mutate over insert.

Reply:
Oh I wasn't aware `mutate()` was possible with maps. does the call have any affect if the value doesn't exist?
Will look into it.